src/serializer.ts

```diff
@@ -84,7 +84,7 @@ export class Serializer {
      * @param clazz The class constructor.
      * @returns An instance of the type `T` with the prototype of `clazz` or one of its registered children.
      */
-    public deserialize<T>(obj: any, clazz?: any): T {
+    public deserialize<T>(obj: any, clazz?: any /* TODO See if we can change this any */): T {
```

Comment:
"change this **any**"
See if we can use something like Instantiable or something